app/src/main/java/pl/futurecollars/invoicing/db/filebased/FileBasedDatabase.java

```diff
@@ -21,8 +21,8 @@
     private final FileService jsonFileService;
     private final FileService idsFileService;
 
-    public FileBasedDatabase(JsonService jsonService) {
-        this.jsonService = jsonService;
+    public FileBasedDatabase() {
+        this.jsonService = new JsonService();
```

Comment:
try to create beans and inject then into FileBasedDatabase. You also can pass FilePathConfig.JSON_FILE when creating beans.